service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchEnableApplicationFlightStep.java

```diff
@@ -0,0 +1,76 @@
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.WsmApplicationKeys;
+import bio.terra.workspace.service.workspace.flight.application.able.AbleEnum;
+import bio.terra.workspace.service.workspace.flight.application.able.ApplicationAbleFlight;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import java.util.List;
+
+public class LaunchEnableApplicationFlightStep implements Step {
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    validateRequiredEntries(
+        context.getInputParameters(),
+        JobMapKeys.AUTH_USER_INFO.getKeyName(),
+        JobMapKeys.REQUEST.getKeyName());
+    validateRequiredEntries(
+        context.getWorkingMap(),
+        WorkspaceFlightMapKeys.APPLICATION_IDS,
+        WorkspaceFlightMapKeys.WsmApplicationKeys.APPLICATION_ABLE_ENUM);
+    final var userRequest =
+        context
+            .getInputParameters()
+            .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    final List<String> applicationIds =
+        context.getWorkingMap().get(WorkspaceFlightMapKeys.APPLICATION_IDS, List.class);
+    final AbleEnum ableEnum =
+        context.getWorkingMap().get(WsmApplicationKeys.APPLICATION_ABLE_ENUM, AbleEnum.class);
+
+    final var destinationWorkspace =
+        context.getInputParameters().get(JobMapKeys.REQUEST.getKeyName(), Workspace.class);
+
+    final Stairway stairway = context.getStairway();
+
+    final FlightMap subflightInputParameters = new FlightMap();
+    subflightInputParameters.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
+    subflightInputParameters.put(WorkspaceFlightMapKeys.APPLICATION_IDS, applicationIds);
+    subflightInputParameters.put(WsmApplicationKeys.APPLICATION_ABLE_ENUM, ableEnum);
+
+    // fields normally set by JobBuilder for identifying jobs
+    subflightInputParameters.put(
+        WorkspaceFlightMapKeys.WORKSPACE_ID, destinationWorkspace.getWorkspaceId().toString());
+    subflightInputParameters.put(
+        JobMapKeys.DESCRIPTION.getKeyName(),
+        String.format(
+            "Enable applications %s in workspace %s",
+            applicationIds, destinationWorkspace.getWorkspaceId()));
+
+    // Build a ApplicationAbleFlight
+    try {
+      stairway.submit(
+          context.getStairway().createFlightId(),
+          ApplicationAbleFlight.class,
+          subflightInputParameters);
+    } catch (DuplicateFlightIdException e) {
+      // exit early if flight is already going
+      return StepResult.getStepResultSuccess();
+    } catch (StairwayException e) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    return StepResult.getStepResultSuccess();
```

Comment:
I talked to Dan and Yu about how to test step undos. Some undo's are tested. Many aren't. But I think it's good practice to test.

Tests can call `.lastStepFailure(true)`. That will fail last step, causing all steps to undo.

Do you mind adding `WorkspaceServiceTest.cloneGcpWorkspaceUndo` ? It will call `.lastStepFailure(true)`. After the flight finishes, it should check:
- Destination workspace doesn't exist
- Cloud context (GCP project) doesn't exist